client/src/test/java/uk/co/blackpepper/bowman/LinkedResourceMethodHandlerTest.java

```diff
@@ -0,0 +1,148 @@
+import uk.co.blackpepper.bowman.annotation.LinkedResource;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.springframework.util.ReflectionUtils.findMethod;
+
+public class LinkedResourceMethodHandlerTest {
+
+    private final ResourceContent resourceContent = new ResourceContent();
+    private final LinkedResourceMethodHandler handler = createHandler();
+
+    @Test
+    public void supportsLinkedResourceSetterIsTrue() {
+        assertThat(handler.supports(findMethod(ResourceContent.class, "setSomething", String.class)), is(true));
+    }
+
+    @Test
+    public void supportsLinkedResourceGetterIsTrue() {
+        assertThat(handler.supports(findMethod(ResourceContent.class, "getSomething")), is(true));
+    }
+
+    @Test
+    public void supportsLinkedResourceSetterWithNoParamIsFalse() {
+        assertThat(handler.supports(findMethod(ResourceContent.class, "setNothing")), is(false));
+    }
+
+    @Test
+    public void supportsLinkedResourceVoidGetterIsFalse() {
+        assertThat(handler.supports(findMethod(ResourceContent.class, "getNothing")), is(false));
+    }
+
+    @Test
+    public void supportsLinkedResourceSetterWithReturnValueIsFalse() {
+        assertThat(handler.supports(findMethod(
+                ResourceContent.class, "setSomethingWithReturn", String.class)), is(false));
+    }
+
+    @Test
+    public void supportsLinkedResourceSetterWithMultipleParamsIsFalse() {
+        assertThat(handler.supports(findMethod(
+                ResourceContent.class, "setSomethingWithExtraParam", String.class, String.class)), is(false));
+    }
+
+    @Test
+    public void supportsLinkedResourceGetterWithParamsIsFalse() {
+        assertThat(handler.supports(findMethod(
+                ResourceContent.class, "getSomethingFromParam", String.class)), is(false));
+    }
+
+    @Test
+    public void supportsNonAnnotatedSetterIsFalse() {
+        assertThat(handler.supports(findMethod(ResourceContent.class, "setSomethingElse", String.class)), is(false));
+    }
+
+    @Test
+    public void supportsNonAnnotatedGetterIsFalse() {
+        assertThat(handler.supports(findMethod(ResourceContent.class, "getSomethingElse")), is(false));
+    }
+
+    @Test
+    public void invokeSetsAndReturnsSameLinkedResource() throws InvocationTargetException, IllegalAccessException {
+        final Method setterMethod = findMethod(ResourceContent.class, "setSomething", String.class);
+        assertThat(setterMethod, is(notNullValue()));
+
+        final String setterValue = "STRING";
+        final Object setterResult = handler.invoke(resourceContent, setterMethod, null, new String[]{setterValue});
+
+        assertThat(setterResult, is(notNullValue()));
+        assertThat(setterResult, instanceOf(String.class));
```

Comment:
Just realised we only need the instanceOf assert, no need for the not null one too.